src/main/java/seedu/financialplanner/investments/WatchList.java

```diff
@@ -154,6 +139,37 @@ public void extractStockInfoFromJSON(JSONObject stock, Stock stockLocal) {
         String dayLow = stock.get("dayLow").toString();
         assert dayLow != null;
         stockLocal.setDayLow(dayLow);
+
+        String timestamp = stock.get("timestamp").toString();
+        long lastUpdated = Long.parseLong(timestamp) * 1000;
+        stockLocal.setLastUpdated(new Date(lastUpdated));
+    }
+
+    public String addStock(String stockCode) throws FinancialPlannerException {
+        if (stocks.size() >= 5) {
+            throw new FinancialPlannerException("Watchlist is full (max 5). Delete a stock to add a new one");
+        }
+        if (stocks.containsKey(stockCode.toUpperCase())) { // should already be uppercase
```

Comment:
Do we still need the toUpperCase method if its alr in uppercase

Reply:
The saved file might be meddled with